### PR TITLE
OnRemoteControlSettings: Initial version of the scripts

### PR DESCRIPTION
**Pull request by dboltovskyi**

### CODE REVIEW

**Comment by dboltovskyi** [comment]
@aderiabin, @Itileda Please review 
Descriptions in scripts will be updated later

**Comment by dboltovskyi** [comment]
@aderiabin, @Itileda Descriptions are added

File: test_scripts/RC/OnRemoteControlSettings/001_SUCCESS_in_case_AUTO_ALLOW_seq_1.lua [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+for _, mod in pairs(modules) do
+  runner.Title("Module: " .. mod)
+  -- set control for App1
+  runner.Step("App1 SetInteriorVehicleData", commonRC.rpcAllowed, { mod, 1, "SetInteriorVehicleData" })
+  for i = 1, #access_modes do
+    runner.Title("Access mode: " .. tostring(access_modes[i]))
+    -- set RA mode
+    runner.Step("Set RA mode", commonRC.defineRAMode, { true, access_modes[i] })
+    -- set control for App2 --> Allowed
+    runner.Step("App2 SetInteriorVehicleData", commonRC.rpcAllowed, { mod, 2, "SetInteriorVehicleData" })
```
Comment (aderiabin) [inline test_scripts/RC/OnRemoteControlSettings/001_SUCCESS_in_case_AUTO_ALLOW_seq_1.lua]: @dboltovskyi  On 2 iteration by accessMode resource is already allocated to 2-nd app. So check of reallocation of resource have no sense. I propose add some steps for allocate resource back to 1-st app before check for new  accessMode.

File: test_scripts/RC/OnRemoteControlSettings/002_SUCCESS_in_case_AUTO_ALLOW_seq_2.lua [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+for _, mod in pairs(modules) do
+  runner.Title("Module: " .. mod)
+  -- set control for App1
+  runner.Step("App1 SetInteriorVehicleData", commonRC.rpcAllowed, { mod, 1, "SetInteriorVehicleData" })
+  for i = 1, #access_modes do
+    runner.Title("Access mode: " .. tostring(access_modes[i]))
+    -- set RA mode
+    runner.Step("Set RA mode", commonRC.defineRAMode, { true, access_modes[i] })
+    -- set control for App2 --> Allowed
+    runner.Step("App2 ButtonPress", commonRC.rpcAllowed, { mod, 2, "ButtonPress" })
```
Comment (aderiabin) [inline test_scripts/RC/OnRemoteControlSettings/002_SUCCESS_in_case_AUTO_ALLOW_seq_2.lua]: @dboltovskyi On 2 iteration by accessMode resource is already allocated to 2-nd app. So check of reallocation of resource have no sense. I propose add some steps for allocate resource back to 1-st app before check for new accessMode.

File: test_scripts/RC/OnRemoteControlSettings/003_SUCCESS_in_case_AUTO_ALLOW_seq_3.lua [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+for _, mod in pairs(modules) do
+  runner.Title("Module: " .. mod)
+  -- set control for App1
+  runner.Step("App1 ButtonPress", commonRC.rpcAllowed, { mod, 1, "ButtonPress" })
+  for i = 1, #access_modes do
+    runner.Title("Access mode: " .. tostring(access_modes[i]))
+    -- set RA mode
+    runner.Step("Set RA mode", commonRC.defineRAMode, { true, access_modes[i] })
+    -- set control for App2 --> Allowed
+    runner.Step("App2 ButtonPress", commonRC.rpcAllowed, { mod, 2, "ButtonPress" })
```
Comment (aderiabin) [inline test_scripts/RC/OnRemoteControlSettings/003_SUCCESS_in_case_AUTO_ALLOW_seq_3.lua]: @dboltovskyi On 2 iteration by accessMode resource is already allocated to 2-nd app. So check of reallocation of resource have no sense. I propose add some steps for allocate resource back to 1-st app before check for new accessMode.

File: test_scripts/RC/OnRemoteControlSettings/004_SUCCESS_in_case_AUTO_ALLOW_seq_4.lua [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+for _, mod in pairs(modules) do
+  runner.Title("Module: " .. mod)
+  -- set control for App1
+  runner.Step("App1 ButtonPress", commonRC.rpcAllowed, { mod, 1, "ButtonPress" })
+  for i = 1, #access_modes do
+    runner.Title("Access mode: " .. tostring(access_modes[i]))
+    -- set RA mode
+    runner.Step("Set RA mode", commonRC.defineRAMode, { true, access_modes[i] })
+    -- set control for App2 --> Allowed
+    runner.Step("App2 SetInteriorVehicleData", commonRC.rpcAllowed, { mod, 2, "SetInteriorVehicleData" })
```
Comment (aderiabin) [inline test_scripts/RC/OnRemoteControlSettings/004_SUCCESS_in_case_AUTO_ALLOW_seq_4.lua]: @dboltovskyi On 2 iteration by accessMode resource is already allocated to 2-nd app. So check of reallocation of resource have no sense. I propose add some steps for allocate resource back to 1-st app before check for new accessMode.

Comment (dboltovskyi) [inline test_scripts/RC/OnRemoteControlSettings/004_SUCCESS_in_case_AUTO_ALLOW_seq_4.lua]: @aderiabin Please check 2bd50a4

File: test_scripts/RC/OnRemoteControlSettings/003_SUCCESS_in_case_AUTO_ALLOW_seq_3.lua (hunk 3 above)
Comment (dboltovskyi) [inline test_scripts/RC/OnRemoteControlSettings/003_SUCCESS_in_case_AUTO_ALLOW_seq_3.lua]: @aderiabin Please check 2bd50a4

File: test_scripts/RC/OnRemoteControlSettings/002_SUCCESS_in_case_AUTO_ALLOW_seq_2.lua (hunk 2 above)
Comment (dboltovskyi) [inline test_scripts/RC/OnRemoteControlSettings/002_SUCCESS_in_case_AUTO_ALLOW_seq_2.lua]: @aderiabin Please check 2bd50a4

File: test_scripts/RC/OnRemoteControlSettings/001_SUCCESS_in_case_AUTO_ALLOW_seq_1.lua (hunk 1 above)
Comment (dboltovskyi) [inline test_scripts/RC/OnRemoteControlSettings/001_SUCCESS_in_case_AUTO_ALLOW_seq_1.lua]: @aderiabin Please check 2bd50a4

File: test_scripts/RC/OnRemoteControlSettings/001_SUCCESS_in_case_AUTO_ALLOW_seq_1.lua [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+end
+
+--[[ Scenario ]]
+runner.Title("Preconditions")
+runner.Step("Clean environment", commonRC.preconditions)
+runner.Step("Start SDL, HMI, connect Mobile, start Session", commonRC.start)
+runner.Step("RAI1, PTU", commonRC.rai_ptu, { ptu_update_func })
+runner.Step("RAI2", commonRC.rai_n, { 2 })
+
+-- App's HMI levels: 1 - BACKGROUND, 2 - FULL
```
Comment (aderiabin) [inline test_scripts/RC/OnRemoteControlSettings/001_SUCCESS_in_case_AUTO_ALLOW_seq_1.lua]: @dboltovskyi HMI level of both apps is NONE here. Please correct comment in code or delete it.

File: test_scripts/RC/OnRemoteControlSettings/002_SUCCESS_in_case_AUTO_ALLOW_seq_2.lua [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+end
+
+--[[ Scenario ]]
+runner.Title("Preconditions")
+runner.Step("Clean environment", commonRC.preconditions)
+runner.Step("Start SDL, HMI, connect Mobile, start Session", commonRC.start)
+runner.Step("RAI1, PTU", commonRC.rai_ptu, { ptu_update_func })
+runner.Step("RAI2", commonRC.rai_n, { 2 })
+
+-- App's HMI levels: 1 - BACKGROUND, 2 - FULL
```
Comment (aderiabin) [inline test_scripts/RC/OnRemoteControlSettings/002_SUCCESS_in_case_AUTO_ALLOW_seq_2.lua]: @dboltovskyi HMI level of both apps is NONE here. Please correct comment in code or delete it.

File: test_scripts/RC/OnRemoteControlSettings/003_SUCCESS_in_case_AUTO_ALLOW_seq_3.lua [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+end
+
+--[[ Scenario ]]
+runner.Title("Preconditions")
+runner.Step("Clean environment", commonRC.preconditions)
+runner.Step("Start SDL, HMI, connect Mobile, start Session", commonRC.start)
+runner.Step("RAI1, PTU", commonRC.rai_ptu, { ptu_update_func })
+runner.Step("RAI2", commonRC.rai_n, { 2 })
+
+-- App's HMI levels: 1 - BACKGROUND, 2 - FULL
```
Comment (aderiabin) [inline test_scripts/RC/OnRemoteControlSettings/003_SUCCESS_in_case_AUTO_ALLOW_seq_3.lua]: @dboltovskyi HMI level of both apps is NONE here. Please correct comment in code or delete it.

File: test_scripts/RC/OnRemoteControlSettings/004_SUCCESS_in_case_AUTO_ALLOW_seq_4.lua [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+end
+
+--[[ Scenario ]]
+runner.Title("Preconditions")
+runner.Step("Clean environment", commonRC.preconditions)
+runner.Step("Start SDL, HMI, connect Mobile, start Session", commonRC.start)
+runner.Step("RAI1, PTU", commonRC.rai_ptu, { ptu_update_func })
+runner.Step("RAI2", commonRC.rai_n, { 2 })
+
+-- App's HMI levels: 1 - BACKGROUND, 2 - FULL
```
Comment (aderiabin) [inline test_scripts/RC/OnRemoteControlSettings/004_SUCCESS_in_case_AUTO_ALLOW_seq_4.lua]: @dboltovskyi HMI level of both apps is NONE here. Please correct comment in code or delete it.

File: test_scripts/RC/OnRemoteControlSettings/015_Empty_data_from_HMI.lua [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+  runner.Title("RA mode Default AUTO_ALLOW")
+  runner.Step("Activate App2", commonRC.activate_app, { 2 })
+  runner.Step("Send OnRemoteControlSettings with invalid data", setRAMode)
+  runner.Step("App2 SetInteriorVehicleData", commonRC.rpcAllowed, { mod, 2, "SetInteriorVehicleData" })
+
+  runner.Title("RA mode AUTO_DENY")
+  runner.Step("Set RA mode", commonRC.defineRAMode, { true, "AUTO_DENY" })
+  runner.Step("Activate App1", commonRC.activate_app, { 1 })
+  runner.Step("Send OnRemoteControlSettings with invalid data", setRAMode)
+  runner.Step("App2 SetInteriorVehicleData", commonRC.rpcDenied, { mod, 1, "SetInteriorVehicleData", "IN_USE" })
```
Comment (aderiabin) [inline test_scripts/RC/OnRemoteControlSettings/015_Empty_data_from_HMI.lua]: @dboltovskyi Please correlate test step description with test function parameters :  
Test step description: "`App2` SetInteriorVehicleData"
Test step function parameters:  { mod, `1`, "SetInteriorVehicleData", "IN_USE" }

The same issue at 58 and 59 lines:
Test step description: "`App2` `ButtonPress` 1st SUCCESS"
Test step function parameters:  { mod, `1`, "`SetInteriorVehicleData`", "IN_USE" }

File: test_sets/rc_OnRemoteControlSettings.txt [hunk 10]
```diff
@@ -1,6 +1,14 @@
...
+./test_scripts/RC/OnRemoteControlSettings/005_REJECTED_in_case_nonFULL_HMI_level.lua
+./test_scripts/RC/OnRemoteControlSettings/006_IN_USE_in_case_2_requests.lua
+./test_scripts/RC/OnRemoteControlSettings/007_IN_USE_in_case_AUTO_DENY.lua
 ./test_scripts/RC/OnRemoteControlSettings/008_Allowed_false.lua
 ./test_scripts/RC/OnRemoteControlSettings/009_Allowed_true_accessMode_AUTO_ALLOW.lua
 ./test_scripts/RC/OnRemoteControlSettings/010_Allowed_true_accessMode_AUTO_DENY.lua
 ./test_scripts/RC/OnRemoteControlSettings/011_Allowed_true_accessMode_ASK_DRIVER.lua
 ./test_scripts/RC/OnRemoteControlSettings/012_Default_accessMode.lua
 ./test_scripts/RC/OnRemoteControlSettings/013_sequence_switches_of_accessMode.lua
+./test_scripts/RC/OnRemoteControlSettings/014_Invalid_data_from_HMI.lua
```
Comment (aderiabin) [inline test_sets/rc_OnRemoteControlSettings.txt]: @dboltovskyi Please add to current test set new test 
(./test_scripts/RC/OnRemoteControlSettings/015_Empty_data_from_HMI.lua)

Comment (dboltovskyi) [inline test_sets/rc_OnRemoteControlSettings.txt]: @aderiabin Please find update in e652c55

File: test_scripts/RC/OnRemoteControlSettings/015_Empty_data_from_HMI.lua (hunk 9 above)
Comment (dboltovskyi) [inline test_scripts/RC/OnRemoteControlSettings/015_Empty_data_from_HMI.lua]: @aderiabin Please find update in e652c55

File: test_scripts/RC/OnRemoteControlSettings/004_SUCCESS_in_case_AUTO_ALLOW_seq_4.lua (hunk 8 above)
Comment (dboltovskyi) [inline test_scripts/RC/OnRemoteControlSettings/004_SUCCESS_in_case_AUTO_ALLOW_seq_4.lua]: @aderiabin Please find update in e652c55

File: test_scripts/RC/OnRemoteControlSettings/003_SUCCESS_in_case_AUTO_ALLOW_seq_3.lua (hunk 7 above)
Comment (dboltovskyi) [inline test_scripts/RC/OnRemoteControlSettings/003_SUCCESS_in_case_AUTO_ALLOW_seq_3.lua]: @aderiabin Please find update in e652c55

File: test_scripts/RC/OnRemoteControlSettings/002_SUCCESS_in_case_AUTO_ALLOW_seq_2.lua (hunk 6 above)
Comment (dboltovskyi) [inline test_scripts/RC/OnRemoteControlSettings/002_SUCCESS_in_case_AUTO_ALLOW_seq_2.lua]: @aderiabin Please find update in e652c55

File: test_scripts/RC/OnRemoteControlSettings/001_SUCCESS_in_case_AUTO_ALLOW_seq_1.lua (hunk 5 above)
Comment (dboltovskyi) [inline test_scripts/RC/OnRemoteControlSettings/001_SUCCESS_in_case_AUTO_ALLOW_seq_1.lua]: @aderiabin Please find update in e652c55

**Pull request merged**